src/python/zquantum/core/interfaces/backend.py

```diff
@@ -114,7 +138,7 @@ def __init__(
         noise_model: Optional = None,
         device_connectivity: Optional[CircuitConnectivity] = None,
     ):
-        self.n_samples = n_samples
+        super().__init__(n_samples)
```

Comment:
I'm just curious what this change actually does :)

Reply:
It lets us borrow the `init` functionality from `QuantumBackend` without rewriting the logic. Before, all QuantumBackend.__init__()` did was `self.n_samples = n_samples`, but now in addition, we set `self.number_of_circuits_run` and `self.number_of_jobs_run`.

Reply:
To clarify, the only reason we rewrite the `__init__` function at all is because we want to add `noise_model` and `device_connectivity` to the function header

Reply:
Sounds good!